torch/csrc/api/include/torch/nn/modules/any.h

```diff
@@ -416,10 +416,19 @@ struct AnyModule::Holder : public AnyModule::Placeholder {
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ AnyModule ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 template <typename ModuleType>
-AnyModule::AnyModule(std::shared_ptr<ModuleType> module)
-    : content_(make_holder(
+AnyModule::AnyModule(std::shared_ptr<ModuleType> module) {
+  // `AnyModule` can only store an `nn::Module` subclass object that provides
+  // a `forward()` method.
+  static_assert(
+      torch::detail::is_module<ModuleType>::value,
+      "Can only store object derived from nn::Module into AnyModule");
+  static_assert(
+      torch::detail::has_forward<ModuleType>::value,
+      "Can only store module with a forward() method into AnyModule");
```

Comment:
We move the `static_assert`s here because `AnyModule` needs to uphold the expectation of storing only `nn::Module` subclass object that provides a `forward()` method.

Reply:
can't the stuff below still be done as member-wise initialization?  Because the only stuff here are static_asserts.